tests/JsonSchema/Tests/Constraints/RequiredPropertyTest.php

```diff
@@ -9,8 +9,60 @@
 
 namespace JsonSchema\Tests\Constraints;
 
+use JsonSchema\Constraints\UndefinedConstraint;
+
 class RequiredPropertyTest extends BaseTestCase
 {
+
+    public function testErrorPropertyIsPopulatedForRequiredIfMissingInInput()
+    {
+        $validator = new UndefinedConstraint();
+        $document = json_decode('{
+            "bar": 42
+        }');
+        $schema = json_decode('{
+            "type": "object",
+            "properties": {
+                "foo": {"type": "number"},
+                "bar": {"type": "number"}
+            },
+            "required": ["foo"]
+        }');
+
+        $validator->check($document, $schema);
+        $error = $validator->getErrors();
+        if(!isset($error[0]['property'])) {
```

Comment:
ah, sorry been away from PHP for a while.  Let me know if latest push works, not the most elegant code but I think it covers any malformed input with a helpful fail message.